Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Tooltips/ToolTipSpawner.cs

```diff
@@ -0,0 +1,350 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections;
+using UnityEngine;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem.Handlers;
+using Microsoft.MixedReality.Toolkit.Core.EventDatum.Input;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem;
+using Microsoft.MixedReality.Toolkit.Core.Definitions.InputSystem;
+using System.Collections.Generic;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.ToolTips
+{
+    /// <summary>
+    /// Add to any Object to spawn ToolTips on tap or on focus, according to preference
+    /// Applies its follow settings to the spawned ToolTip's ToolTipConnector component
+    /// </summary>
+    public class ToolTipSpawner : MonoBehaviour, IMixedRealityInputHandler, IMixedRealityFocusHandler
+    {
+        private enum VanishType
+        {
+            VanishOnFocusExit,
+            VanishOnTap,
+        }
+
+        private enum AppearType
+        {
+            AppearOnFocusEnter,
+            AppearOnTap,
+        }
+
+        public enum RemainType
+        {
+            Indefinite,
+            Timeout,
+        }
+
+        [SerializeField]
+        private Vector3 defaultDimensions = new Vector3(0.182f, 0.028f, 1.0f);
+
+        [SerializeField]
+        private bool showBackground = true;
+
+        [SerializeField]
+        private bool showOutline = false;
+
+        [SerializeField]
+        private bool showConnector = true;
+
+        [SerializeField]
+        private AppearType appearType = AppearType.AppearOnFocusEnter;
+
+        [SerializeField]
+        private VanishType vanishType = VanishType.VanishOnFocusExit;
+
+        [SerializeField]
+        private RemainType remainType = RemainType.Timeout;
+
+        [SerializeField]
+        [Tooltip("The action that will be used for when to spawn or toggle the tooltip.")]
+        private MixedRealityInputAction tooltipToggleAction = MixedRealityInputAction.None;
+
+        [SerializeField]
+        [Range(0f, 5f)]
+        private float appearDelay = 0.0f;
```

Comment:
Is this something that should be allowed to be set in code?

Reply:
Please provide tooltip attributes to fields that appear in the inspector.

Reply:
Added tooltips in PR against his branch.